File: regions/system/library/dart/outbox/lib/src/store.dart
import 'message.dart';

/// アウトボックスメッセージの永続化インターフェース。
abstract class OutboxStore {
  Future<void> save(OutboxMessage msg);
  Future<List<OutboxMessage>> fetchPending(int limit);
  Future<void> update(OutboxMessage msg);
  Future<int> deleteDelivered(int olderThanDays);
}

/// メッセージを外部に送信するインターフェース。
abstract class OutboxPublisher {
  Future<void> publish(OutboxMessage msg);
}

File: regions/system/library/python/outbox/src/k1s0_outbox/store.py
"""OutboxStore 抽象基底クラス"""

from __future__ import annotations

import uuid
from abc import ABC, abstractmethod

from .models import OutboxMessage


class OutboxStore(ABC):
    """Outbox ストレージ抽象基底クラス。"""

    @abstractmethod
    async def save(self, message: OutboxMessage) -> None:
        """メッセージを保存する。"""
        ...

    @abstractmethod
    async def fetch_pending(self, limit: int = 100) -> list[OutboxMessage]:
        """PENDING 状態のメッセージを取得する。"""
        ...

    @abstractmethod
    async def mark_published(self, message_id: uuid.UUID) -> None:
        """メッセージを PUBLISHED に更新する。"""
        ...

    @abstractmethod
    async def mark_failed(self, message_id: uuid.UUID, error: str) -> None:
        """メッセージを FAILED に更新する。"""
        ...

    @abstractmethod
    async def increment_retry(self, message_id: uuid.UUID) -> None:
        """リトライカウントをインクリメントする。"""
        ...
